pkg/service/restore/tables_worker.go

```diff
@@ -227,11 +228,26 @@ func (w *tablesWorker) stageRestoreData(ctx context.Context) error {
 
 			pr, err := w.newRunProgress(ctx, hi, b)
 			if err != nil {
-				return errors.Wrap(err, "create new run progress")
+				err = multierr.Append(errors.Wrap(err, "create new run progress"), bd.ReportFailure(hi.Host, b))
+				w.logger.Error(ctx, "Failed to create new run progress",
+					"host", hi.Host,
+					"error", err)
+				if ctx.Err() != nil {
+					return err
+				}
+				continue
 			}
 			if err := w.restoreBatch(ctx, b, pr); err != nil {
-				return errors.Wrap(err, "restore batch")
+				err = multierr.Append(errors.Wrap(err, "restore batch"), bd.ReportFailure(hi.Host, b))
+				w.logger.Error(ctx, "Failed to restore batch",
+					"host", hi.Host,
+					"error", err)
+				if ctx.Err() != nil {
+					return err
+				}
+				continue
```

Comment:
You wrote on PR description:
```
After batch failed to be restored, SM cleans the upload dir, so that there are no left overs there.
```

Where is this part exactly ?

Reply:
It has always been like that, it's a part of the `tablesWorker.cleanupRunProgress` method called at the end of `tablesWorker.restoreBatch` method.

Reply:
I understand the integration-test added with this PR tests this behavior, as it fails to copypaths or to call for load&stream, right ?

Reply:
The test does not ensure that.
It might happen that failed node picks up another batch from the same table, but it might not happen.